Modules/System/AL Rest Client/src/ALRestClient.Codeunit.al

```diff
@@ -0,0 +1,339 @@
+/// <summary>Provides functionality to easily work with the HttpClient object.</summary>
+codeunit 2350 "AL Rest Client"
+{
+    var
+        ALRestClientImpl: Codeunit "AL Rest Client Impl.";
+        IsInitialized: Boolean;
+
+    #region Initialization
+    /// <summary>Initializes the AL Rest Client with anonymous authentication.</summary>
+    procedure Initialize()
+    var
+        HttpAuthenticationAnonymous: Codeunit "Http Authentication Anonymous";
+    begin
+        Initialize(HttpAuthenticationAnonymous);
+    end;
+
+    /// <summary>Initializes the AL Rest Client with the given authentication.</summary>
+    /// <param name="HttpAuthentication">The authentication to use.</param>
+    procedure Initialize(HttpAuthentication: Interface "Http Authentication")
+    begin
+        ALRestClientImpl.Initialize();
+        ALRestClientImpl.SetAuthentication(HttpAuthentication);
+        IsInitialized := true;
+    end;
+
+    /// <summary>Sets a default request header.</summary>
+    /// <remarks>Default request headers will be added to every request that is sent with this AL Rest Client instance
+    /// The AL Rest Client will be initialized if it was not initialized before.</remarks>
+    /// <param name="Name">The name of the request header.</param>
+    /// <param name="Value">The header of request header.</param>
+    procedure SetDefaultRequestHeader(Name: Text; Value: Text)
+    begin
+        if not IsInitialized then
+            Initialize();
+
+        ALRestClientImpl.SetDefaultRequestHeader(Name, Value);
+    end;
+
+    /// <summary>Sets the base address of the AL Rest Client.</summary>
+    /// <remarks>The base address will be used for every request that is sent with this AL Rest Client instance.
+    /// Calls to the Get, Post, Patch, Put and Delete methods must  use a relative path which will be appended to the base address.
+    /// The AL Rest Client will be initialized if it was not initialized before.</remarks>
+    /// <param name="Url">The base address to use.</param>
+    procedure SetBaseAddress(Url: Text)
+    begin
+        if not IsInitialized then
+            Initialize();
+
+        ALRestClientImpl.SetBaseAddress(Url);
```

Comment:
I recall the HttpClient throwing a runtime error if some of the "default" properties were set after a request had been made, for example BaseAddress, Timeout, the Default Headers, Authorization, etc. but I didn't see any tests validating this.
Should the module clear the client if it's setting these properties after Init+Send? Or do you know if this has been somehow addressed under-the-hood?